python/seldon_core/storage.py

```diff
@@ -23,15 +23,18 @@
 import os
 import re
 from urllib.parse import urlparse
-from azure.storage.blob import BlockBlobService
 from minio import Minio
-from seldon_core.imports_helper import _GCS_PRESENT
+from seldon_core.imports_helper import _AZURE_PRESENT, _GCS_PRESENT
 from seldon_core.utils import getenv
 
 if _GCS_PRESENT:
     from google.auth import exceptions
     from google.cloud import storage
 
+if _AZURE_PRESENT:
+    from azure.storage.blob import BlockBlobService
```

Comment:
Just wonder, we make these imports dependent on the library presence but we don't check bellow when they are used.

Could it lead to some obscure `undefined name` errors? 
Would it be good idea to check bellow for values of `_GCS_PRESENT` / `_AZURE_PRESENT`.

I guess it's because we do not want to modify `storage.py` as we gonna drop our copy of it and use upstream soon?

Reply:
Yeah, that's the main reason. To keep these changes simple. 

We do print a log message already if Azure or any of the other optional dependencies are not there, which is handled here: 

https://github.com/adriangonz/seldon-core/blob/0095ba5a39e98caff2b86c2a3f7cec422279d1c4/python/seldon_core/imports_helper.py#L65-L84

I also imagine that the logic to check the value of `_AZURE_PRESENT` inside `Storage.py` could be a bit more nuanced. For example, what happens if we try to use it but it's not there, should we just raise an exception or fail silently?

Reply:
Hmm... If it fail silenty it'd mean it didn't download required files so we will get errors somewhere else that may be harder to debug.

Right now it will raise `NameError`  (I guess) -> the only better suggestion I could have would be to raise some custom `DependencyError` with more informant error message. 

Anyway, that would be only potential improvement and as current logic was already present for `_GCS_PRESENT` it shouldn't stop this from being merged.